### DISCUSSION TITLE
Auto-reindex of bag files

### DISCUSSION THREAD

**Discussion by valschmidt** [foxglove/community#14] [Q&A]
On our operational robots it is pretty common to have bag files that didn’t get closed cleanly and need to be reindexed. It happens for a whole host of reasons, and I think this will be common with other users. Of course we can fix this ourselves, but It would be a nice feature if the foxglove import process would attempt to reindex a bag file if it is required before reporting a failed import. This feature would be handy for both the foxglove online data service and foxglove-studio.

**Comment by amacneil** [comment]
Thanks for the feedback.

We err on the side of caution right now by rejecting corrupted or uncleanly closed files. We could definitely add support for running a recovery script on our side, but we would need to be super clear with an alert that you may have data loss in that file. I'm not sure if we would want to do this automatically, or offer it as a manual step for files which we detect as corrupt.

Do you notice any other issues viewing/playing these files in other tools today? (e.g. `rosbag play` with RViz?) Do they load from your local filesystem in Foxglove Studio ok, or do you get an error there too?

**Reply by geetesh** [reply]
Thanks for this awesome tool. I have the same request. 
I often use macOS with M1 chip and foxglove studio is just so convenient when digging through bags. Sometimes bags are corrupt and then I have to setup ROS environment just to run rosbag reindex.
Reindexing the bag does allow us to play it successfully in rosbag/rviz/foxglove studio but it would be super awesome to have studio offer the option to reindex without having to go CLI route OR even installing ROS.
Thanks again

**Reply by valschmidt** [reply]
@amacneil, I'm sorry to be so slow to respond to your Q. 
Yes, I think other tools fail to execute when a file needs reindexing. But that too is a hassle. 

It's not clear to me what the danger is of re-indexing a file if it needs it. 

Seems like a good options is to inform the user that the file needs reindexing and inquire if the user would like the file reindexed in place or reindexed into a temporary file. This would allow the user to load a read-only file, or, if they want the file fixed once and for all, to let Foxglove Studio fix it.
